lib/membrane_rtsp/server/conn.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Membrane.RTSP.Server.Conn do
+  @moduledoc """
+  A module representing a client-server connection
+  """
+
+  use GenServer
+
+  require Logger
+
+  alias Membrane.RTSP.Server.Logic
+
+  @max_request_size 1_000_000
+
+  @spec start(map()) :: GenServer.on_start()
+  def start(state) do
+    GenServer.start(__MODULE__, state)
+  end
+
+  @impl true
+  def init(config) do
+    state = %Logic.State{
+      socket: config.socket,
+      request_handler: config.handler,
+      request_handler_state: config.handler.handle_open_connection(config.socket),
+      rtp_socket: config.udp_rtp_socket,
+      rtcp_socket: config.udp_rtcp_socket,
+      session_timeout: config.session_timeout
+    }
+
+    {:ok, state, {:continue, :process_client_requests}}
+  end
+
+  @impl true
+  def handle_continue(:process_client_requests, state) do
+    do_process_client_requests(state)
+    state.request_handler.handle_closed_connection(state.request_handler_state)
```

Comment:
Just to confirm, we don't care whether the connection got closed normally or due to an error?

Reply:
I thought about passing the reason to the callback, however I don't think the handler will do anything with this value, the end result would be the same for the handler: free resources in case `TEARDOWN` is not received.